idioms/ffi-intro.md

```diff
@@ -0,0 +1,5 @@
+# FFI Idioms
+
+Writing FFI code is an entire course in itself. However, there are several idioms here that can act a pointers, and avoid traps for inexperienced users of unsafe Rust.
+
+This section contains idioms that may be useful when doing FFI.
```

Comment:
Proposing to in-repo link here the sections as a small TOC if it works as an intro to those files